src/components/actionBar/actionBar/ActionBar.tsx

```diff
@@ -29,15 +28,73 @@ interface IActionBarProps {
 const ActionBar: React.FC<IActionBarProps> = ({
     showFilterSort,
 }: IActionBarProps) => {
+    const tomorrow = addDays(new Date(), 1);
+    const [searchParams] = useSearchParams();
     const [startLocation, setStartLocation] = useState<ILocationOptions | null>(
         null
     );
+
     const [stopLocation, setStopLocation] = useState<ILocationOptions | null>(
         null
     );
-
+    const [tripDate, setTripDate] = useState<Date | null>(tomorrow);
+    const [locOptions, setLocOptions] = useState<ILocationOptions[]>([]);
     const { t } = useTranslation('actionBar');
     const navigate = useNavigate();
+    const dispatch = useDispatch();
+
+    const [toggle, setToggle] = useState(false);
+
+    const originParam = searchParams.get('originID');
+    const destinationParam = searchParams.get('destinationID');
+    const tripDateParam = searchParams.get('tripDate');
+
+    // setting origin and destination from query params
+    const setParamOptions = () => {
+        const originlocation = locOptions.find((opt) => {
+            return opt.id == Number(originParam);
+        });
+        const destinationlocation = locOptions.find((opt) => {
+            return opt.id === Number(destinationParam);
+        });
+
+        if (originlocation) {
+            setStartLocation(originlocation);
+        }
+        if (destinationlocation) {
+            setStopLocation(destinationlocation);
+        }
+        if (tripDate && tripDateParam) {
+            const date = tripDate.toUTCString();
+            const convDate = new Date(date);
+            setTripDate(convDate);
+        }
+    };
+
+    // querying the locations
+    const getLocOptions = async () => {
+        try {
+            const loc = await getLocations();
+            const converterLoc = loc.map((locObj) => {
+                return { id: Number(locObj.id), label: locObj.name };
+            });
+            setLocOptions(() => converterLoc);
+        } catch (err) {
+            // show error
+            console.log('There is an error', err);
+        }
+    };
+
+    useEffect(() => {
+        getLocOptions().catch(() => {
+            console.log('couldnt fetch location into locOptions');
```

Comment:
is it required?

Reply:
ESLint rule to add catch, removing the try-catch instead to avoid redundancy:https://github.com/nijin-26/bus-ticket-booking-react/pull/24/commits/949f9eb00945de8723e70618e8816603f116682a